code/lib/cli/src/automigrate/fixes/eslint-plugin.ts

```diff
@@ -89,16 +90,23 @@ export const eslintPlugin: Fix<EslintPluginRunOptions> = {
       `);
     }
 
-    const eslint = await readConfig(eslintFile);
-    logger.info(`✅ Configuring eslint rules in ${eslint.fileName}`);
-
+    logger.info(`✅ Adding Storybook plugin to ${eslintFile}`);
     if (!dryRun) {
-      logger.info(`✅ Adding Storybook to extends list`);
-      const extendsConfig = eslint.getFieldValue(['extends']) || [];
-      const existingConfigValue = Array.isArray(extendsConfig) ? extendsConfig : [extendsConfig];
-      eslint.setFieldValue(['extends'], [...existingConfigValue, 'plugin:storybook/recommended']);
-
-      await writeConfig(eslint);
+      if (eslintFile.endsWith('json')) {
+        const eslintConfig = (await readJson(eslintFile)) as { extends?: string[] };
+        const existingConfigValue = Array.isArray(eslintConfig.extends)
+          ? eslintConfig.extends
+          : [eslintConfig.extends];
+        eslintConfig.extends = [...(existingConfigValue || []), 'plugin:storybook/recommended'];
+        await writeJson(eslintFile, eslintConfig, { spaces: 2 });
```

Comment:
Should we try to infer the amount of spaces from the existing eslint file? otherwise we risk changing the whole file for one line, with an annoying diff to follow.

Reply:
We can use this:
https://www.npmjs.com/package/detect-indent

Reply:
If you have a good solution for that, I'm all in! We have a similar issue in the other bit, for js/ts files (in other automigrations).
The `writeConfig` from csf-tools will write file in a way that could impact the formatting, but we assume the users will have prettier in place, so that would not be much of a big deal 🤞